new-lamassu-admin/src/pages/LoyaltyPanel/CouponCodesModal.js

```diff
@@ -0,0 +1,110 @@
+import { makeStyles } from '@material-ui/core/styles'
+import { Form, Formik, Field } from 'formik'
+import * as R from 'ramda'
+import React from 'react'
+import * as Yup from 'yup'
+
+import Modal from 'src/components/Modal'
+import { Tooltip } from 'src/components/Tooltip'
+import { Button } from 'src/components/buttons'
+import { TextInput, NumberInput } from 'src/components/inputs/formik'
+import { H3, TL1, P } from 'src/components/typography'
+
+import styles from './CouponCodes.styles'
+
+const useStyles = makeStyles(styles)
+
+const initialValues = {
+  code: '',
+  discount: ''
+}
+
+const validationSchema = Yup.object().shape({
+  code: Yup.string()
+    .required()
+    .trim()
+    .max(25),
+  discount: Yup.number()
+    .required()
+    .min(0)
+    .max(100)
+})
+
+const CouponCodesModal = ({ showModal, onClose, errorMsg, addCoupon }) => {
+  const classes = useStyles()
+
+  const handleAddCoupon = (code, discount) => {
+    addCoupon(R.toUpper(code), parseInt(discount))
+  }
+
+  return (
+    <>
+      {showModal && (
+        <Modal
+          title="Add coupon code discount"
+          closeOnBackdropClick={true}
+          width={600}
+          height={500}
+          handleClose={onClose}
+          open={true}>
+          <Formik
+            initialValues={initialValues}
+            validationSchema={validationSchema}
+            onSubmit={({ code, discount }, { resetForm }) => {
+              handleAddCoupon(code, discount)
+              resetForm()
```

Comment:
What happens if the save function fails? Why the need to resetForm here?

Reply:
Solved